lib/ansible/modules/monitoring/grafana_datasource.py

```diff
@@ -313,7 +313,7 @@ def grafana_datasource_exists(module, grafana_url, name, headers):
     r, info = fetch_url(module, '%s/api/datasources/name/%s' % (grafana_url, name), headers=headers, method='GET')
     if info['status'] == 200:
         datasource_exists = True
-        ds = json.loads(r.read())
+        ds = json.loads(r.read().decode('utf-8'))
```

Comment:
there are other locations in the file as well where such a fix is required

Reply:
Use 

```
ds= json.loads(to_text(r.read(), errors='surrogate_or_strict'))
```

instead of `decode()`

Reply:
OK, i've replaced all `json.loads(r.read())` with `json.loads(to_text(r.read(), errors='surrogate_or_strict'))` and tested, it works.